### ISSUE TITLE
CPU temperature not working

### ISSUE THREAD

**Issue by Liszur** [flightlessmango/MangoHud#147]
Hey,
I got the new 0.3.5 version and had a great installation process on a freshly installed Manjaro 20 and MangoHud seems to work fine, but i can't get CPU temperature readings because it drops an error: `MANGOHUD: Could not find cpu temp sensor location`.

Here is my lshw about my mobo and cpu:

>  *-core
>        description: Motherboard
>        product: 304Ah
>        vendor: Hewlett-Packard
>        physical id: 0
>        serial: CZC1168XVY
>      *-firmware
>           description: BIOS
>           vendor: Hewlett-Packard
>           physical id: 1
>           version: 786H1 v01.13
>           date: 07/14/2011
>           size: 128KiB
>           capacity: 8128KiB
>           capabilities: pci pnp upgrade shadowing cdboot bootselect edd int5printscreen int9keyboard int14serial int17printer acpi usb ls120boot zipboot biosbootspecification netboot
>      *-cpu
>           description: CPU
>           product: Intel(R) Core(TM) i5 CPU         760  @ 2.80GHz
>           vendor: Intel Corp.
>           physical id: 5
>           bus info: cpu@0
>           version: Intel(R) Core(TM) i5 CPU         760  @ 2.80GHz
>           slot: XU1 PROCESSOR
>           size: 2757MHz
>           capacity: 2933MHz
>           width: 64 bits
>           clock: 133MHz
>           capabilities: x86-64 fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx rdtscp constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni dtes64 monitor ds_cpl smx est tm2 ssse3 cx16 xtpr pdcm sse4_1 sse4_2 popcnt lahf_lm pti ssbd ibrs ibpb stibp dtherm ida flush_l1d cpufreq
>           configuration: cores=4 enabledcores=4 threads=4
>

**Comment by flightlessmango** [comment]
cpu temp requires the `lm_sensors` package

**Comment by Liszur** [comment]
Yeah, i also thought for this and installed it before posting but problem remained the same.

**Comment by jackun** [comment]
Does `sensors` itself show it? What does  `cat /sys/class/hwmon/hwmon*/name` say?

**Comment by Liszur** [comment]
> Does `sensors` itself show it? What does `cat /sys/class/hwmon/hwmon*/name` say?

sensors shows all 4 core temp and `cat /sys/class/hwmon/hwmon*/name` writes `coretemp`

**Comment by jackun** [comment]
Ok, probably is not `temp1_input` then. Run `ls /sys/class/hwmon/hwmon*/*_input`

**Comment by Liszur** [comment]
You are right, i have cpu temps from /sys/class/hwmon/hwmon0/temp2_input to /sys/class/hwmon/hwmon0/temp5_input

**Comment by utherbone** [comment]
I was about to make a topic about it, well and I'm going through the same problem.
What can I do for MangoHud to display the CPU temperature?

**Command:** sensors
```
$ sensors
atk0110-acpi-0
Adapter: ACPI interface
Vcore Voltage:        1.38 V  (min =  +0.85 V, max =  +1.60 V)
 +3.3 Voltage:        3.10 V  (min =  +2.97 V, max =  +3.63 V)
 +5 Voltage:          4.78 V  (min =  +4.50 V, max =  +5.50 V)
 +12 Voltage:        12.04 V  (min = +10.20 V, max = +13.80 V)
SB Voltage:           1.07 V  (min =  +0.90 V, max =  +1.29 V)
SB2 Voltage:          1.65 V  (min =  +1.22 V, max =  +1.82 V)
FSBT Voltage:         1.39 V  (min =  +1.00 V, max =  +1.48 V)
DRAM Voltage:         2.24 V  (min =  +1.50 V, max =  +2.28 V)
DRAM Ref Voltage:     1.12 V  (min =  +0.75 V, max =  +1.14 V)
NB Voltage:           1.58 V  (min =  +1.04 V, max =  +1.56 V)
CPU FAN Speed:      1259 RPM  (min =  600 RPM, max = 7200 RPM)
CHASSIS1 FAN Speed:    0 RPM  (min =  600 RPM, max = 7200 RPM)
CHASSIS2 FAN Speed:    0 RPM  (min =  600 RPM, max = 7200 RPM)
CHASSIS3 FAN Speed: 2625 RPM  (min = 1800 RPM, max = 7200 RPM)
PWR FAN Speed:         0 RPM  (min = 1800 RPM, max = 7200 RPM)
OPT_Fan1 FAN Speed:    0 RPM  (min =  800 RPM, max = 7200 RPM)
OPT_Fan2 FAN Speed: 1360 RPM  (min =  800 RPM, max = 7200 RPM)
OPT_Fan3 FAN Speed:    0 RPM  (min =  800 RPM, max = 7200 RPM)
CPU Temperature:     +34.0°C  (high = +35.0°C, crit = +90.0°C)
MB Temperature:      +33.0°C  (high = +30.0°C, crit = +90.0°C)
NB Temperature:      +46.0°C  (high = +45.0°C, crit = +95.0°C)
SB Temperature:      +36.0°C  (high = +45.0°C, crit = +95.0°C)
OPT1 Temperature:     +0.0°C  (high = +45.0°C, crit = +95.0°C)
OPT2 Temperature:     +0.0°C  (high = +45.0°C, crit = +95.0°C)
OPT3 Temperature:     +0.0°C  (high = +45.0°C, crit = +95.0°C)

coretemp-isa-0000
Adapter: ISA adapter
Core 0:       +47.0°C  (high = +76.0°C, crit = +100.0°C)
Core 1:       +43.0°C  (high = +76.0°C, crit = +100.0°C)
Core 2:       +49.0°C  (high = +76.0°C, crit = +100.0°C)
Core 3:       +49.0°C  (high = +76.0°C, crit = +100.0°C)
```
**Command:** cat /sys/class/hwmon/hwmon*/name
```
$ cat /sys/class/hwmon/hwmon*/name
atk0110
coretemp
```
**Command:** ls /sys/class/hwmon/hwmon*/*_input
```
$ ls /sys/class/hwmon/hwmon*/*_input
/sys/class/hwmon/hwmon0/fan1_input  /sys/class/hwmon/hwmon0/in7_input
/sys/class/hwmon/hwmon0/fan2_input  /sys/class/hwmon/hwmon0/in8_input
/sys/class/hwmon/hwmon0/fan3_input  /sys/class/hwmon/hwmon0/in9_input
/sys/class/hwmon/hwmon0/fan4_input  /sys/class/hwmon/hwmon0/temp1_input
/sys/class/hwmon/hwmon0/fan5_input  /sys/class/hwmon/hwmon0/temp2_input
/sys/class/hwmon/hwmon0/fan6_input  /sys/class/hwmon/hwmon0/temp3_input
/sys/class/hwmon/hwmon0/fan7_input  /sys/class/hwmon/hwmon0/temp4_input
/sys/class/hwmon/hwmon0/fan8_input  /sys/class/hwmon/hwmon0/temp5_input
/sys/class/hwmon/hwmon0/in0_input   /sys/class/hwmon/hwmon0/temp6_input
/sys/class/hwmon/hwmon0/in1_input   /sys/class/hwmon/hwmon0/temp7_input
/sys/class/hwmon/hwmon0/in2_input   /sys/class/hwmon/hwmon1/temp2_input
/sys/class/hwmon/hwmon0/in3_input   /sys/class/hwmon/hwmon1/temp3_input
/sys/class/hwmon/hwmon0/in4_input   /sys/class/hwmon/hwmon1/temp4_input
/sys/class/hwmon/hwmon0/in5_input   /sys/class/hwmon/hwmon1/temp5_input
/sys/class/hwmon/hwmon0/in6_input
```
Greetings,
Gabriel (Utherbone)

**Comment by utherbone** [comment]
Me CPU is Core 2 Quad Q9550
In this report [insue] (https://github.com/flightlessmango/MangoHud/issues/60) a user had the CPU core 2 duo and suggested a build from a developer branch, I tested that version but without success.

Something I can do to see my CPU temperature without the change in the MangoHud code?

**Comment by flightlessmango** [comment]
This is possibly fixed in develop d93a21589e7faf1310c42acb2cb4dad4a01a32f0

**Comment by flightlessmango** [comment]
If you're still having issues after this fix, feel free to open up the issue again
